test/test_integration_check.py

```diff
@@ -0,0 +1,192 @@
+        self.assertIn(output, "false\n")
+
+    @integration_test(True)
+    def test_check_non_existent_privilege_on_non_permitted_resource_with_permitted_specified_role(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['check', '-i', 'dev:policy:somepolicy', '-p', 'create', '-r', 'dev:user:permitted_user'])
+        self.assertIn(output, "false\n")
+
+    @integration_test(True)
+    def test_check_existent_privilege_on_non_permitted_resource_with_non_permitted_specified_role(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['check', '-i', 'dev:policy:somepolicy', '-p', 'read', '-r', 'dev:user:non_permitted_user'])
+        self.assertIn(output, "false\n")
+
+    @integration_test(True)
+    def test_check_non_existent_privilege_on_permitted_resource_with_non_permitted_specified_role(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['check', '-i', 'dev:variable:somevariable', '-p', 'execute', '-r', 'dev:layer:non_permitted_layer'])
+        self.assertIn(output, "false\n")
+
+    @integration_test(True)
+    def test_check_non_existent_privilege_on_non_permitted_resource_with_non_permitted_specified_role(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['check', '-i', 'dev:webservice:somewebservice', '-p', 'update', '-r', 'dev:host:non_permitted_host'])
+        self.assertIn(output, "false\n")
+
+    '''
+    Validates that when the user isn't logged in and makes a request,
+    they are prompted to login first and then the command is executed
+    '''
+    @integration_test()
+    @patch('builtins.input', return_value='admin')
+    def test_check_without_user_logged_in_prompts_login_and_performs_check_with_valid_commands_default_role(self, mock_input):
+        # TEST_ENV is set to False so we will purposely be prompted to login
+        os.environ['TEST_ENV'] = 'False'
+        try:
+            utils.delete_credentials()
+        except OSError:
+            pass
+
+        with patch('getpass.getpass', return_value=self.client_params.env_api_key):
+            output = self.invoke_cli(self.cli_auth_params,
+                                     ['check', '-i', 'dev:variable:somevariable', '-p', 'read'])
+
+            self.assertIn(LOGIN_IS_REQUIRED, output)
+            self.assertIn("Successfully logged in to Conjur", output)
+            expected_output = 'To start using the CLI, log in to Conjur\nSuccessfully logged in to Conjur\ntrue\n'
+            self.assertIn(output, expected_output)
+        os.environ['TEST_ENV'] = 'True'
+
+    @integration_test()
+    @patch('builtins.input', return_value='admin')
+    def test_check_without_user_logged_in_prompts_login_and_performs_check_with_valid_commands_specified_role(self, mock_input):
```

Comment:
We only need one of these tests to confirm that the command gets run after the login prompt. The optional `role` argument should be sufficiently covered by your other tests.

Reply:
removed extra test. kept the default/standard role one